### PR TITLE
refactor: Add delegated function impls for `Labels` and `Annotations`

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/188

~Final merge is blocked until https://github.com/Kobzol/rust-delegate/pull/64 is officially released.~ Version `0.12.0` was released on 2023-12-22.

### CODE REVIEW

**Review by sbernauer — DISMISSED** [review]
Whup whup

File: src/kvp/label/mod.rs [hunk 1]
```diff
@@ -305,4 +278,25 @@ impl Labels {
 
         Ok(labels)
     }
+
+    delegate! {
```
Comment (lfrancke) [inline src/kvp/label/mod.rs:282]: Maybe it's just me but it's not obvious what this is doing. I have no idea what "delegate!" is.
If in doubt, add a comment

Comment (Techassi) [reply src/kvp/label/mod.rs:282]: Sure. I will add a comment. It basically forwards the impl to the inner type.

Comment (lfrancke) [reply src/kvp/label/mod.rs:282]: You don't have to start a new PR for this.
It was more a general remark: Help your future self and other future people with logging and comments.

Comments should (in my opinion) mostly about the "why" not the "what", sometimes - if it's not obvious - about the "how this works"

Comment (Techassi) [reply src/kvp/label/mod.rs:286]: This should now be resolved by 20274e7

**Review by sbernauer — APPROVED** [review]

**Review by lfrancke — APPROVED** [review]

**Pull request merged**